test/testsuites/opsrcdeletetests.go

```diff
@@ -0,0 +1,47 @@
+package testsuites
+
+import (
+	"testing"
+
+	"github.com/operator-framework/operator-marketplace/test/helpers"
+	"github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/stretchr/testify/assert"
+)
+
+// DeleteOpSrc tests that the correct cleanup occurs when an OpSrc is deleted
+func DeleteOpSrc(t *testing.T) {
+	t.Run("delete-operator-source", testDeleteOpSrc)
+}
+
+// testDeleteOpSrc ensures that after deleting an OperatorSource that the
+// objects created as a result are deleted
+func testDeleteOpSrc(t *testing.T) {
+	ctx := test.NewTestCtx(t)
+	defer ctx.Cleanup()
+
+	// Get global framework variables
+	client := test.Global.Client
+
+	// Get test namespace
+	namespace, err := test.NewTestCtx(t).GetNamespace()
+	assert.NoError(t, err, "Could not get namespace.")
+
+	testOperatorSource := helpers.CreateOperatorSourceDefinition(namespace)
+
+	// Create the OperatorSource with no cleanup options.
+	err = helpers.CreateRuntimeObjectNoCleanup(client, testOperatorSource)
+	assert.NoError(t, err, "Could not create operator source.")
+
+	// Check for the datastore CatalogSourceConfig and its child resources.
+	err = helpers.CheckCatalogSourceConfigAndChildResourcesCreated(test.Global.Client, testOperatorSource.Name, namespace, namespace)
+	assert.NoError(t, err, "Could not ensure that CatalogSourceConfig and its cild resources were deleted")
```

Comment:
Same here, if we don't have a properly initialized `opsrc` then we can abort the test, I assume.

Reply:
If this applies to https://github.com/operator-framework/operator-marketplace/pull/177, please mention it on that PR too.

Reply:
This is a good point. It also seems like, for basically every test we have that requires the previous step to complete before the next, almost every since call should be `require` instead of `assert`. In this case, I just updated the pr to do just that. We should probably ensure that this is also the case for https://github.com/operator-framework/operator-marketplace/pull/177

Reply:
A test usually has three regions:
```
func Test(t *testing.T) {
  // Arrange
  // This is where the dependencies of function under test are resolved. If we fail to initialize here 
  // we need to abort the test. require is more appropriate. 
  require.NoError()

  // Act:
  // This is where we invoke the function under test.

  // Assert:
  // This is where we assert that output from act meets expectation. So assert is more appropriate 
  // here. if I have N asserts, I can see all the failed expectations.
  assert.NoError()
}
```

Reply:
@tkashem In the case of unit tests I would agree, but these are functional tests and we aren't just setting up objects and calling an API or a set of APIs and checking the result, each step (Arrange, Act, or Assert) is an actual functional API call. We aren't mocking anything, so we can't assume any result at any time.

Reply:
This is applicable to all type of tests - for this test `delete opsrc`:
Arrange or prerequisites: A properly initialized opsrc object.
Act: delete the opsrc object.
Assert: Make sure all resources are cleaned up.

So yeah, even if it is functional, we do need to setup expectations so that we can assert on them.